### ISSUE TITLE
privatize Buffer constants

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-js#1660]
Some constants are showing up in the API docs that should be `@private`:

![image](https://cloud.githubusercontent.com/assets/98601/10860715/bdb9e130-7f29-11e5-89ec-7a51dc7291be.png)

**Comment by lucaswoj** [comment]
Oops! Sorry :disappointed:

**Comment by lucaswoj** [comment]
Also, I think I've been seeing some doc generation warning messages generated by `Buffer`, I'll tackle that in the same PR.

**Comment by lucaswoj** [comment]
Interesting -- these actually *are* marked as `@private` 

```js
/**
 * @enum {string} BufferType
 * @private
 * @readonly
 */
Buffer.BufferType = ...
```

I'm going to do some more digging.

**Comment by lucaswoj** [comment]
Looks like this was caused by a bug in `documentationjs` that has been fixed as of v3.0. I'm not sure if I should uncodument these properties (`/**` -> `/*`) or wait for the upgrade. I think @jfirebaugh is working on an upgrade PR now.

**Comment by lucaswoj** [comment]
ref #1663
